application/controllers/grader.php

```diff
@@ -43,15 +50,13 @@ public function run()
 		while (TRUE)
 		{
 			sleep(1);
-
-			$this->check_in();
-			if ( ! $this->has_new_submission())
-				continue;
-
-			$submission = $this->get_new_submission();
-			$this->grade_submission($submission);
-			
-			$this->logger->log('grader', 'Waiting for new submission...');
+			$this->check_in($number);
```

Comment:
The function `check_in` below doesn't have the parameter `number`, but this function call passes it.
Are you sure this is working?